tests/api/test_chip.cpp

```diff
@@ -206,14 +213,23 @@ TEST(ApiChipTest, SpecifyLegalDeassertRiscResetOnCore) {
     umd_cluster->l1_membar(chip_core_coord.chip, "LARGE_WRITE_TLB");
 
     uint32_t soft_reset_reg_addr = 0xFFB121B0;
-    uint32_t risc_reset_val;
-    umd_cluster->read_from_device(&risc_reset_val, chip_core_coord, soft_reset_reg_addr, sizeof(uint32_t), "REG_TLB");
-    EXPECT_EQ(static_cast<uint32_t>(deassert_val), risc_reset_val);
+    uint32_t deassert_reset_val;
+    umd_cluster->read_from_device(&deassert_reset_val, chip_core_coord, soft_reset_reg_addr, sizeof(uint32_t), "REG_TLB");
+    EXPECT_EQ(static_cast<uint32_t>(deassert_val), deassert_reset_val);
+
+    TensixSoftResetOptions assert_val = TENSIX_ASSERT_SOFT_RESET & 
+        static_cast<TensixSoftResetOptions>(~std::underlying_type<TensixSoftResetOptions>::type(TensixSoftResetOptions::BRISC));
+    umd_cluster->assert_risc_reset_at_core(chip_core_coord, assert_val);
+    umd_cluster->l1_membar(chip_core_coord.chip, "LARGE_WRITE_TLB");
+
+    uint32_t assert_reset_val;
+    umd_cluster->read_from_device(&assert_reset_val, chip_core_coord, soft_reset_reg_addr, sizeof(uint32_t), "REG_TLB");
+    EXPECT_EQ(static_cast<uint32_t>(assert_val), assert_reset_val);
 }
 
-// // This tests puts a specific core into reset and then specifies an illegal deassert value
-// // It reads back the risc reset reg to validate that reset reg is in a legal state
-TEST(ApiChipTest, SpecifyIllegalDeassertRiscResetOnCore) {
+// This tests puts a specific core into reset and then specifies illegal deassert and assert values
+// It reads back the risc reset reg to validate that reset reg is in a legal state
+TEST(ApiChipTest, SpecifyIllegalAssertDeassertRiscResetOnCore) {
```

Comment:
It may be that something in this test (or previous tests) drives cards to a bad state? Or maybe this works non-deterministicaly? I'm seeing some failures on my branch without changes in this PR: https://github.com/tenstorrent/tt-umd/actions/runs/11889008268/job/33124886089?pr=310

Reply:
Yeah I’m not sure what’s going on with these tests. It really just is register writes and reads  I’m not sure if I’m missing something here

Reply:
Another failure: https://github.com/tenstorrent/tt-umd/actions/runs/11966959645/job/33363528724?pr=323